src/lib/carplay/PVCarPlay.android.ts

```diff
@@ -1,55 +1,250 @@
+            (episode?.liveItem?.start && readableDate(episode.liveItem.start)) ||
+            (episode.pubDate && readableDate(episode.pubDate)) ||
+            ''
+          return {
+            title: episode.title || translate('Untitled Episode'),
+            subtitle: pubDate,
+            playable: '0',
+            iconUri: episode.imageUrl || podcast.imageUrl || undefined,
+            mediaId: `${MediaKeys.Episode}-${podcast.id}-${index}`
+          }
+        })
+      })
+    }
+  }
+}
+
+export const handlePlayRemoteMediaId = async (mediaId: string) => {
+  if (mediaId.startsWith(MediaKeys.Queue)) {
+    const episodeId = mediaId.substring(MediaKeys.Queue.length + 1)
+    const foundEpisode = getQueue().filter((episode) => episode.episodeId === episodeId)
+    if (foundEpisode.length === 0) {
+      errorLogger(
+        _fileName,
+        'handlePlayRemoteMediaId',
+        `[Android Auto] ${mediaId} no longer exists in the current Queue.`
+      )
+      return
+    }
+    const convertedEpisode = convertNowPlayingItemToEpisode(foundEpisode[0])
+    loadEpisodeInPlayer(convertedEpisode, convertedEpisode.podcast)
+  } else if (mediaId.startsWith(MediaKeys.History)) {
+    const episodeId = mediaId.substring(MediaKeys.History.length + 1)
+    const foundEpisode = (await getHistory()).filter((episode) => episode.episodeId === episodeId)
+    if (foundEpisode.length === 0) {
+      errorLogger(
+        _fileName,
+        'handlePlayRemoteMediaId',
+        `[Android Auto] ${mediaId} no longer exists in the current History.`
+      )
+      return
+    }
+    const convertedEpisode = convertNowPlayingItemToEpisode(foundEpisode[0])
+    await loadEpisodeInPlayer(convertedEpisode, convertedEpisode.podcast)
+    refreshHistory()
+  } else if (mediaId.startsWith(MediaKeys.Episode)) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [_, podcastId, index] = /^Episode-(.+)-(\d+)$/.exec(mediaId)!
+    try {
+      loadEpisodeInPlayer(cachedEpisodes[podcastId][Number(index)], cachedPodcasts[podcastId])
+    } catch (e) {
+      errorLogger(_fileName, 'handlePlayRemoteMediaId', `[Android Auto] ${mediaId}, ${podcastId}, ${index}, ${e}`)
+    }
+  } else {
+    errorLogger(_fileName, 'handlePlayRemoteMediaId', `[Android Auto] ${mediaId} format is not supported.`)
+  }
+}
+
+/* Initialize */
+
+export const onAppInitialized = () => {
+  const { PVAndroidAutoModule } = NativeModules
+  PVAndroidAutoModule.turnOffShowWhenLocked()
+  handleAndroidAutoPodcastsUpdate()
+  // example dialog to prompt for android auto draw over apps permission.
+  PVAndroidAutoModule.getDrawOverAppsPermission().then((enabled: boolean) => {
+    if (enabled) return
+    Alert.alert(translate('Android Auto Permission Title'), translate('Android Auto Permission Body'), [
+      { text: 'Cancel', style: 'cancel' },
+      { text: 'OK', onPress: () => PVAndroidAutoModule.askDrawOverAppsPermission() }
+    ])
+  })
```

Comment:
This looks good but I did get the Alert as soon as I launched the app on my phone without being connected to my car. Is there a way you think we can show this to the users the first time they connect their device to Android Auto instead always on launch?

Reply:
actually that wont work:
The "draw over other app" permission is required when the app is fully closed and connected to android auto. Otherwise users have to keep podverse open while connecting to the car for the first time for this alert to show on the phone and be enabled, while there is no way to display this alert on the car head unit screen.